### PR TITLE
Add french translation

### PR DESCRIPTION
**Pull request by BertrandClosset**
# Context
- No issues was opened about it but I wanted to help by translating the project in french

# Summary of Changes

- add `fr`folder with all the `yml` files translated in french  in `config/locales`
- changed the value of `config.i18n.available_locales` to add the french language in the `application.rb`file

# Checklist

- [ ] Tested Mobile Responsiveness
- [ ] Added Unit Tests
- [ ] CI Passes
- [ ] Deploys to Heroku on test Correctly (Maintainers will handle)
- [ ] Added Documentation (Service and Code when required)

### CODE REVIEW

**Comment by DeeDeeG** [comment]
Wow, thank you for this translation!

I can give technical feedback to make sure this works as expected. And we would also like to get another native speaker or two to review the translation, if possible.

Anyways, merci beaucoup! (I don't speak much French, but I do know that much!)

**Comment by BertrandClosset** [comment]
You're welcome, it's always a pleasure to help !
I can ask some friends to take a look at the translation.
On the technical side, I installed the application (i just followed the instructions which are in the contributing file) and I test with the french version, it was working well.
I don't have knowledge in Ruby but I want to try to help you on this side (I saw the issue about the language switcher, I will take a look and give it a try on my side)

File: config/locales/fr/business_info.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+fr:
+  business_info:
+    title: "Questions fréquentes des organisations et entreprises à propos de Refuge Restrooms"
+    p1header: "1) Qu'est ce que Refuge Restrooms ?"
+    p1: "REFUGE est une application web visant à fournir un accès sécurisé à des toilettes aux personnes transgenres, intersexuées et non conformes dans le genre. En lire plus à propos de nous sur notre <a href='/about'>page À Propos</a>."
+    p2header: "2) Comment mon établissement est-il entré dans cette liste ?"
+    p2: "Quelqu'un a utilisé vos toilettes et voulait soumettre votre emplacement en tant que ressource. Toutes nos données sont fournies par l'utilisateur et soumises par l'utilisateur."
+    p3header: "3) Comment puis-je être un bon hôte ?"
+    p3: "Merci d'avoir posé la question! Souvent, les visiteurs de REFUGE espèrent faire un arrêt rapide et être sur notre chemin, et comme tout autre visiteur, potentiellement soutenir votre entreprise ou votre emplacement pendant que nous sommes là. Le personnel amical et de bonnes expériences peuvent apporter un peu d'amour pour votre espace dans la section des commentaires de l'application (plus d'informations ci-dessous)"
```
[the remark this answers is not in the document]
Comment (TiteiKo) [reply config/locales/fr/business_info.yml]: "Friendly staff and good experiences can go a long way in getting some love for your location in the reviews section of the app (more on that below!) " => "Un personnel amical et une bonne expérice peut faire beaucoup pour que la section des commentaires sur votre emplacement se remplisse de mots gentils"

File: config/locales/fr/business_info.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    p4header: "4) Why are safe restrooms an issue?"
+    p4:
+      first: "Aller aux toilettes es une activité quotidienne qui peut être difficile à effectuer pour les membres de notre communauté -- en particulier quand ils se rendent dans des lieux étrangers à leurs habitudes. C'est à ce moment que notre projet carthographie intervient."
```
Comment (TiteiKo) [inline config/locales/fr/business_info.yml]: "que notre projet carthographie intervient" => "que notre projet de cartographie intervient"

File: config/locales/fr/business_info.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      second: "Selon l'<a href='http://www.ustranssurvey.org/'>US Trans Survey</a>:"
+      list1: "59% des répondants évitent d'utiliser les toilettes par peur de confrontation"
+      list2: "31% évitent de manger ou boire à l'extérieur pour eviter d'utiliser les toilettes"
+      list3: "24%  ont été questionnés sur leur présence aux toilettes"
+      list4: "12% ont été harcelés, attaqués, ou agressés sexuellement dans les toilettes"
+      list5: "9% se sont vus refuser l'accès aux toilettes"
+      list6: "8% ont développé un problème rénal ou une infection des voies urinaires pour avoir éviter l'utilisation des toilettes"
+    p5header: "5) Que signifient les avis?"
+    p5: "Les avis aident les visiteurs à partager leurs expériences les uns avec les autres sur l'emplacement. Nous encourageons les visiteurs à partager autant d'informations que nous avons sur chaque emplacement de la carte."
```
Comment (TiteiKo) [inline config/locales/fr/business_info.yml]: Nous encourageons les visiteurs à partager autant d'informations que possible sur chaque emplacement de la carte.

File: config/locales/fr/contacts.fr.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+fr:
+  contacts:
+    new:
+      request-edit-for-restroom: "Demande d'edition pour Toilettes %{restroom_name}"
```
Comment (TiteiKo) [inline config/locales/fr/contacts.fr.yml]: Demande d'édition des toilettes %{restroom_name}

File: config/locales/fr/footer.fr.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  layouts:
+    footer:
+      aria-labels:
+        github: 'Refuge Restrooms sur GitHub'
+        twitter: 'Refuge Restrooms sur twitter'
+        facebook: 'Refuge Restrooms sur Facebook'
+        tumblr: 'Le blog de Refuge Restrooms sur tumblr'
+        email: 'Email Refuge Restrooms'
+      refuge-restrooms-is-open-source: 'Refuge Restrooms est open source.'
+      code-on-github: 'Code sur github.'
```
Comment (TiteiKo) [inline config/locales/fr/footer.fr.yml]: Code source sur GitHub

File: config/locales/fr/navigation.fr.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+fr:
+  layouts:
+    navigation:
+      toggle-navigation-button-label: 'Basculer en mode navigation'
+      submit-a-new-restroom-hyperlink-label: 'Soumettre de nouvelles toilettes'
```
Comment (TiteiKo) [inline config/locales/fr/navigation.fr.yml]: Ajouter de nouveaux toilettes

File: config/locales/fr/restroom.fr.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
+fr:
+  activerecord:
+    attributes:
+      restroom:
+        # only overridden attributes here, defaults work as expected
+        # (e.g. database field 'city' will be appear as 'City')
+        name: "Emplacement/Lieu/Nom de l'entreprise"
+        street: 'Adresse'
+        state: 'État/Région'
+        access: 'Accessible?'
```
Comment (TiteiKo) [inline config/locales/fr/restroom.fr.yml]: Space before ?

File: config/locales/fr/restroom.fr.yml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    enter_location: 'Entrez une adresse'
+
+  restroom:
+    add_new: 'Soumettre des toilettes à notre base de données'
+    preview: 'Aperçu'
+    required: '* indique un champ requis'
+    guess_location: 'Trouver ma position actuelle'
+    directions: 'Indications'
+    directions_hint: "(ex: troisième étage au fond, à coté des cabines d'essayage...)"
+    comments_hint: '(ex: vous devez être un client, faites comme si vous cherchiez quelque chose...)'
```
Comment (TiteiKo) [inline config/locales/fr/restroom.fr.yml]: vous devez être client

File: config/locales/fr/restroom.fr.yml [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+      unrated: 'Pas encore noté'
+    upvote: "J'ai adoré!"
+    downvote: 'Pas sécurisé'
+    confirm:
+      upvote: 'Êtes vous sûr de vouloir donner un vote positif pour ces toilettes ?'
+      downvote: 'Êtes vous sûr de vouloir donner un vote négatif pour ces toilettes ?'
+    flash:
+      field: "Un champ requis n'a pas été complété. Veuillez revérifier vos données et soumettre le formulaire à nouveau."
+      unexpected: 'Nous avons rencontré un problème inattendu, veuillez nous contacter si ce problème persiste.'
+      upvoteerror: "Une erreur inattendu s'est produite durant le vote positif de cette publication."
```
Comment (TiteiKo) [inline config/locales/fr/restroom.fr.yml]: Une erreur inattendue s'est produite durant votre vote positif pour cette publication

File: config/locales/fr/restroom.fr.yml [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+      upvotesuccess: 'Ces toilettes ont reçu un vote positif! Merci pour votre contribution à notre communauté.'
+      downvoteerror: "Une erreur inattendu s'est produite durant le vote négatif de cette publication."
```
Comment (TiteiKo) [inline config/locales/fr/restroom.fr.yml]: Une erreur inattendue s'est produite durant votre vote négatif pour cette publication

File: config/locales/fr/restroom.fr.yml [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+      downvotesuccess: 'Ces toilettes ont reçu un vote négatif! Merci pour votre contribution à notre communauté.'
```
Comment (TiteiKo) [inline config/locales/fr/restroom.fr.yml]: Space before !

File: config/locales/fr/restroom.fr.yml [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    upvote: "J'ai adoré!"
+    downvote: 'Pas sécurisé'
+    confirm:
+      upvote: 'Êtes vous sûr de vouloir donner un vote positif pour ces toilettes ?'
+      downvote: 'Êtes vous sûr de vouloir donner un vote négatif pour ces toilettes ?'
+    flash:
+      field: "Un champ requis n'a pas été complété. Veuillez revérifier vos données et soumettre le formulaire à nouveau."
+      unexpected: 'Nous avons rencontré un problème inattendu, veuillez nous contacter si ce problème persiste.'
+      upvoteerror: "Une erreur inattendu s'est produite durant le vote positif de cette publication."
+      upvotesuccess: 'Ces toilettes ont reçu un vote positif! Merci pour votre contribution à notre communauté.'
```
Comment (TiteiKo) [inline config/locales/fr/restroom.fr.yml]: Space before !

File: config/locales/fr/restroom.fr.yml [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+      downvoteerror: "Une erreur inattendu s'est produite durant le vote négatif de cette publication."
+      downvotesuccess: 'Ces toilettes ont reçu un vote négatif! Merci pour votre contribution à notre communauté.'
+      new: 'De nouvelles toilettes ont été ajoutées pour %{name}.'
+      updated: 'Ces toilettes ont été mises à jour'
+      deleted: 'Ces toilettes ont été supprimées'
+      searcherror: "Une erreur s'est produite lors de la recherche de votre position."
+    contact:
+      contactlisting: "Contactez nous à propos de cette publication!"
```
Comment (TiteiKo) [inline config/locales/fr/restroom.fr.yml]: Space before !

File: config/locales/fr/text_msg.fr.yml [hunk 14]
```diff
@@ -0,0 +1,7 @@
+fr:
+  pages:
+     text:
+        title: "Obtenez les toilettes les plus proches par SMS"
```
Comment (TiteiKo) [inline config/locales/fr/text_msg.fr.yml]: Recevez les toilettes les plus proches par SMS

File: config/locales/fr/text_msg.fr.yml [hunk 15]
```diff
@@ -0,0 +1,7 @@
...
+        intro: "Si vous avez besoin de connaitre les toilettes sécurisées les plus proches <em>tout de suite</em>, vous pouvez les recevoir par SMS sur n'importe quelle téléphone."
```
Comment (TiteiKo) [inline config/locales/fr/text_msg.fr.yml]: n'importe quel téléphone

**Comment by DeeDeeG** [comment]
@TiteiKo,
Thank you for the detailed comments, this is a very thorough review.

File: config/locales/fr/about.fr.yml [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+      second: "<strong>Deuxièmement:</strong> Diffuser le message. Tweet. Facebook. Tumblr. Blog. Quoi que vous fassiez, faites-le. Faites connaître aux gens cette application."
+      third: "<strong>Troisièmement:</strong> Si vous savez coder, visitez GitHub et informez-nous d'un bug, suggérez une amélioration, ou ajoutez un peu de code et aidez le projet. REFUGE est open source et nous ne pouvons pas le faire sans vous."
+      fourth: "<strong>Quatrièmement: </strong> Faites un don. Gardez les yeux ouverts pour une campagne de crowdfunding à venir pour financer une partie de la technologie dont nous avons besoin et pour payer un peu nos fabuleux concepteurs et ingénieurs pour leur travail acharné et inlassable. Ils ont travaillé gratuitement pour vous offrir ce service et nous ne voulons pas que quiconque travaille gratuitement. La plupart des membres de l'équipe de base sont transgenres et sous-employés en ce moment."
+    p6header: "Pourquoi mon entreprise ou organisation est-elle répertoriée ?"
+    p6: "En savoir plus sur notre <a href='/business_info'>Business FAQ</a>!"
+    contribute: 'Forkez nous et contribuez sur GitHub!'
```
[the remark this answers is not in the document]
Comment (DeeDeeG) [reply config/locales/fr/about.fr.yml]: FWIW: "Fork us on GitHub" sounds strange and a bit sexual in English, IMO! (Maybe just "Contribute on GitHub!" would be better for the English?) But this phrase is promoted by GitHub, and among people familiar with GitHub I'd say it's just accepted. Maybe for a general audience "Contribuez sur GitHub !" is better...

**Comment by BertrandClosset** [comment]
@TiteiKo Thanks for the review, there was a lot of stupids mistakes that i didn't see before, I'm currently fixing everything

File: config/locales/fr/about.fr.yml [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+    p4header: "Quelle est l'importance des toilettes et pourquoi avons nous besoin de cette application ?"
+    p4: "L'un des plus grands champs de bataille où se déroule chaque jour la lutte pour les droits des transgenres est celui des toilettes: il semble que toutes les deux semaines un enfant transgenre soit au centre d'un reportage national parce qu'il utilise les toilettes assignées au sexe qu'il identifie. Évidemment, nous croyons que toute personne transgenre devrait avoir le droit d'utiliser les toilettes qu'elle souhaite. Pourtant, nous réalisons également que malgré les victoires législatives de ces dernières années concernant l'utilisation des toilettes, de nombreux individus transgenres sont toujours confrontés au harcèlement verbal et physique simplement pour avoir utilisé les toilettes. Personne ne devrait être confronté à cela - et c'est pourquoi nous avons créé REFUGE."
+    p5header: 'Que pouvons nous faire pour aider?'
+    p5:
+      first: "<strong>Premièrement:</strong> Référencer des toilettes. La taille de la base de données dépend de votre participation. Plus il y a de toilettes référencées, plus l'application sera complète et aura de la valeure."
```
[the remark this answers is not in the document]
Comment (BertrandClosset) [reply config/locales/fr/about.fr.yml]: yeah i spent some time thinking on this one, but i think you're right, it's better with the numbers

**Comment by mi-wood** [comment]
Thank you @BertrandClosset and @TiteiKo! I'm cool with merging this now.

@BertrandClosset do you have a twitter handle I can tag you with when I deploy this?

**Comment by TiteiKo** [comment]
@DeeDeeG @BertrandClosset @mi-wood You're welcome, glad I could help a bit an important cause. Don't hesitate to ping me on twitter if/when needed 👍

**Pull request merged**